examples/lighting-app/esp32/main/main.cpp

```diff
@@ -113,6 +117,24 @@ static void InitServer(intptr_t context)
 
     DeviceCallbacksDelegate::Instance().SetAppDelegate(&sAppDeviceCallbacksDelegate);
     Esp32AppServer::Init(); // Init ZCL Data Model and CHIP App Server AND Initialize device attestation config
+
+#if CONFIG_ENABLE_ESP_INSIGHTS_TRACE
+    esp_insights_config_t config = {
+        .log_type = ESP_DIAG_LOG_TYPE_ERROR | ESP_DIAG_LOG_TYPE_WARNING | ESP_DIAG_LOG_TYPE_EVENT,
+        .auth_key = insights_auth_key_start,
+    };
+
+    esp_err_t ret = esp_insights_init(&config);
+
+    if (ret != ESP_OK)
+    {
+        ESP_LOGE(TAG, "Failed to initialize ESP Insights, err:0x%x", ret);
+    }
+#endif
+
+#if CONFIG_ENABLE_ESP_INSIGHTS_SYSTEM_STATS
```

Comment:
Can we move this block inside `#if CONFIG_ENABLE_ESP_INSIGHTS_TRACE`?

Reply:
Made the suggested change.